app/controllers/metrics_controller.rb

```diff
@@ -0,0 +1,359 @@
+    @team = @participant.team # team method in AssignmentParticipant return the AssignmentTeam of this participant
+    @team_id = @team.id
+
+    # retrieve github data and store in the instance variables defined above
+    retrieve_github_data
+
+    # get each PR's status info
+    query_all_merge_statuses
+
+    #@authors = @authors.keys # only keep the author name info
+    @dates = @dates.keys.sort # only keep the date info and sort
+
+    @participants = get_data_for_list_submissions(@team)
+
+    # Create database entry for basic statistics. These data are queried later by view_team in grades (the heatgrid)
+    @authors.each do |author|
+      # Check to see if this author is a member of the expertiza dev team. This COULD be done with a query,
+      # But will only work if the authenticated user running the query has push access to the github repository
+      # (Per Github API security rules)
+      unless LOCAL_ENV["COLLABORATORS"].include? author[1]
+        # If author is a student, keep the commit data and store the total as a Metric
+        data_object = {}
+        data_object[:author] = author[0] # Github Name
+        data_object[:email] = author[1] # Github Email
+        data_object[:commits] = @parsed_data[author[0]].values.inject(0) {|sum, value| sum += value} #Sum of commits
+        create_github_metric(@team_id, author[1], data_object[:commits])
+      end
+    end
+  end
+
+
+
+
+  ##################### Process Links and Branch according to Pull Request or Repo ############################
+  # For a single assignment team, process the submitted links, determine whether they are pull request links or
+  # repository links, and branch accordingly to query github for the data from the type of link found. The github API
+  # works differently and has different available data for pull requests and repositories.
+  def retrieve_github_data
+    team_links = @team.hyperlinks # all links that a team submitted
+    pull_links = team_links.select do |link|
+      link.match(/pull/) && link.match(/github.com/) # all links that contain both pull and github.com
+    end
+    if !pull_links.empty? # have pull links, retrieve pull request info
+      query_all_pull_requests(pull_links)
+    else # retrieve repo info if no PR is submitted
+    repo_links = team_links.select do |link|
+      link.match(/github.com/)
+    end
+    retrieve_repository_data(repo_links)
+    end
+  end
+
+
+  ############### Handling of Pull Request Links #####################
+
+  # Iterate through all pull request links, and initiate the github graphql API queries for each link. Then, call
+  # parse_pull_request_data to process the returned data from each link
+  def query_all_pull_requests(pull_links)
+    pull_links.each do |hyperlink|
+      submission_hyperlink_tokens = hyperlink.split('/') # parse the link
+      hyperlink_data = {}
+      hyperlink_data["pull_request_number"] = submission_hyperlink_tokens[6] # 1858
```

Comment:
could you add a comment explaining the api better or at least linking inside the code to some documentation. the 6 and various numbers is kind of confusing